apps/server/src/modules/account/services/account-idm.service.spec.ts

```diff
@@ -166,6 +166,31 @@ describe('AccountService Integration', () => {
 			expect(updateSpy).not.toHaveBeenCalled();
 			expect(createSpy).toHaveBeenCalled();
 
+			expect(ret).toBeDefined();
+			expect(ret).toMatchObject<Partial<AccountDto>>({
+				id: mockIdmAccount.attRefTechnicalId,
+				idmReferenceId: mockIdmAccount.id,
+				createdAt: mockIdmAccount.createdDate,
+				updatedAt: mockIdmAccount.createdDate,
+				username: mockIdmAccount.username,
+			});
+		});
+		it('should create a new account on update error', async () => {
```

Comment:
The idea is to follow a self-healing approach here. The IDM should try to create accounts which were not created earlier. I updated the code to catch the discovery of an IDM account only, so that we do not attempt to create an account in case the (real intend) update failed.